parameter_expansion/tests/test_pe.py

```diff
@@ -79,6 +79,78 @@
     ),
 }
 
+substring_test_cases = {
+    # string,               parameter, result
+    "-${parameter:0:2}-": (
+        "aa/bb/cc",
+        "-aa-",
+    ),
+    "-${parameter:3:2}-": (
+        "aa/bb/cc",
+        "-bb-",
+    ),
+    "-${parameter:6}-": (
+        "aa/bb/cc",
+        "-cc-",
+    ),
+}
+
+replace_test_cases = {
+    # string,               parameter, result
+    "-${parameter/aa/bb}-": (
+        "aa/bb/cc",
+        "-bb/bb/cc-",
+    ),
+    "-${parameter/aa/}-": (
+        "aa/bb/cc",
+        "-/bb/cc-",
+    ),
+    "-${parameter/aa/zz}-": (
+        "aa/bb/aa",
+        "-zz/bb/aa-",
+    ),
+    "-${parameter//aa/zz}-": (
+        "aa/bb/aa",
+        "-zz/bb/zz-",
+    ),
+}
+
+simple_test_cases = {
+    # string,               env, result
+    "-${parameter/$aa/$bb}-": (
+        dict(
```

Comment:
Why `dict()` instead of a literal dict?

Reply:
I feel it makes its stand out more clearly as being a mapping of environment variables... but I am fine either way